### ISSUE TITLE
Is svelte-preprocess-unocss compatible with unocss-preset-theme?

### ISSUE THREAD

**Issue by CapitalMarkets** [jacob-8/svelte-scoped-uno#1]
Hi, I've been using [unocss-preset-theme](https://github.com/Dunqing/unocss-preset-theme) with [temp-s-p-u](https://www.npmjs.com/package/temp-s-p-u) with no issues for a while now.

I had noticed at some point that temp-s-p-u or something downstream to it throws the following error for unocss > 0.48.5 (0.49.0 upwards) but was happy to stick with this version and push on:
```
> Cannot read properties of undefined (reading 'join')
    at Object.match (/home/user/Projects/uxi-web-ui/node_modules/unocss/node_modules/@unocss/preset-mini/dist/shared/preset-mini.52ab4b0f.cjs:347:74)
    at UnoGenerator.matchVariants (file:///home/user/Projects/uxi-web-ui/node_modules/temp-s-p-u/node_modules/@unocss/core/dist/index.mjs:636:25)
    at UnoGenerator.parseToken (file:///home/user/Projects/uxi-web-ui/node_modules/temp-s-p-u/node_modules/@unocss/core/dist/index.mjs:484:26)
    at file:///home/user/Projects/uxi-web-ui/node_modules/temp-s-p-u/node_modules/@unocss/core/dist/index.mjs:520:34
    at Array.map (<anonymous>)
    at UnoGenerator.generate (file:///home/user/Projects/uxi-web-ui/node_modules/temp-s-p-u/node_modules/@unocss/core/dist/index.mjs:517:46)
    at Object.getCSS (/home/user/Projects/uxi-web-ui/node_modules/unocss-preset-theme/dist/index.cjs:113:51)
    at file:///home/user/Projects/uxi-web-ui/node_modules/temp-s-p-u/node_modules/@unocss/core/dist/index.mjs:550:118
    at Array.map (<anonymous>)
    at file:///home/user/Projects/uxi-web-ui/node_modules/temp-s-p-u/node_modules/@unocss/core/dist/index.mjs:550:93
    at Array.map (<anonymous>)
    at file:///home/user/Projects/uxi-web-ui/node_modules/temp-s-p-u/node_modules/@unocss/core/dist/index.mjs:547:57
    at UnoGenerator.generate (file:///home/user/Projects/uxi-web-ui/node_modules/temp-s-p-u/node_modules/@unocss/core/dist/index.mjs:557:7)
    at async transformSvelteSFC (file:///home/user/Projects/uxi-web-ui/node_modules/temp-s-p-u/dist/index.mjs:2120:27)
    at async markup (file:///home/user/Projects/uxi-web-ui/node_modules/temp-s-p-u/dist/index.mjs:2252:22)
    at async process_markup (file:///home/user/Projects/uxi-web-ui/node_modules/svelte/compiler.mjs:44034:23)
    at async preprocess (file:///home/user/Projects/uxi-web-ui/node_modules/svelte/compiler.mjs:44064:30)
    at async process_file (file:///home/user/Projects/uxi-web-ui/node_modules/@sveltejs/package/src/index.js:200:27)
    at async Module.build (file:///home/user/Projects/uxi-web-ui/node_modules/@sveltejs/package/src/index.js:61:3)
    at async file:///home/user/Projects/uxi-web-ui/node_modules/@sveltejs/package/src/cli.js:27:4

```

I then happened upon a [strange issue](https://github.com/Dunqing/unocss-preset-theme/issues/9) that ended up being related to the name of one of my components (Checkbox.svelte) that I originally thought was with unocss-preset-theme which was failing to write out scoped CSS classes/variables but that I now know is either down to temp-s-p-u, unocss or other dependency clashing with my Checkbox.svelte (is that even supposed to be possible given imports/namespaces :thinking:)

Anyway yesterday, I discovered this project which I figured would grab bits of temp-s-p-u and other improvements you might have made integrating scoped svelte styles in unocss, and decided to give it a go.

unocss is removed from the package.json as I understand svelte-preprocess-unocss takes care of the dependency version.

svelte-package is running the build and there is no vite specific config.

Basically with svelte-preprocess-unocss configured, unocss-preset-theme doesn't appear to run or generate the css style/var settings for my themes.

I've included two minimal examples to illustrate my before (temp-s-p-u) vs my after (svelte-preprocess-unocss):

[temp-s-p-u__unocss-preset-theme.zip](https://github.com/jacob-8/svelte-scoped-uno/files/11179322/temp-s-p-u__unocss-preset-theme.zip)
[svelte-scoped-uno__unocss-preset-theme.zip](https://github.com/jacob-8/svelte-scoped-uno/files/11179323/svelte-scoped-uno__unocss-preset-theme.zip)

_components_
Checkbox - broken even before
Checkbox2 - expected to generate as before
Newbox - expected to generate as before
Newbox2 - expected to generate as before
